### PR TITLE
feat(integrations): Add frontend analytics

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Integrations paged viewed
 - Integrations installation modal opened
 - Integration details (configure) viewed

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx [hunk 1]
```diff
@@ -19,6 +20,12 @@ export default class OrganizationIntegrations extends AsyncComponent {
   reloadOnVisible = true;
   shouldReloadOnVisible = true;
 
+  componentDidMount() {
+    analytics('integrations.index_viewed', {
+      org_id: this.props.params.orgId,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: This stores the slug not the numeric ID, that's wrong.

File: src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.jsx [hunk 2]
```diff
@@ -23,6 +24,13 @@ export default class InstalledIntegration extends React.Component {
     onReinstallIntegration: PropTypes.func.isRequired,
   };
 
+  componentDidMount() {
+    analytics('integrations.details_viewed', {
+      org_id: this.props.orgId,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.jsx]: Also stores the slug and not the numeric ID

File: src/sentry/static/sentry/app/components/modals/integrationDetailsModal.jsx [hunk 3]
```diff
@@ -45,9 +46,16 @@ class IntegrationDetailsModal extends React.Component {
     closeModal: PropTypes.func.isRequired,
     onAddIntegration: PropTypes.func.isRequired,
     provider: PropTypes.object.isRequired,
-    organization: SentryTypes.Organization,
+    organization: SentryTypes.Organization.isRequired,
   };
 
+  componentDidMount() {
+    analytics('integrations.install_modal_opened', {
+      org_id: this.props.organization.id,
```
Comment (adhiraj) [inline src/sentry/static/sentry/app/components/modals/integrationDetailsModal.jsx]: Gotta parseInt I think?

File: src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx (hunk 1 above)
Comment (adhiraj) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: So we should change it then?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx]: Yes, was just leaving this as a note for myself.

**Review by adhiraj — APPROVED** [review]

**Pull request merged**